### ISSUE TITLE
Kodi Connect - movie upgrade doesn't update kodi library

### ISSUE THREAD

**Issue by zadowsmash** [Radarr/Radarr#2771]
**Description:**

Radarr downloads an update of an existing movie. Sends a notification to Kodi to update library and the library scan fails. Radarr sends the existing movie file name to kodi instead of the updated one and kodi can't find the path as it is no longer valid.

In the case of the debug logs below, the original file name was 101.Dalmatians.1996.avi where the new file name is 101.Dalmatians.1996.mkv

**Radarr Version:**

0.2.0.1067

**Mono Version:**

5.10.1.47 (tarball Tue Apr 17 15:12:35 UTC 2018)

**Debug Logs:**

```
07:52:17.557 T:140213671048960  NOTICE: VideoInfoScanner: Starting scan ..                                                                 
07:52:18.337 T:140213671048960 WARNING: Process directory 'smb://192.168.0.11/Movies/101.Dalmatians.1996/101.Dalmatians.1996.avi' does not 
exist - skipping scan and clean.           
```

**Comment by zadowsmash** [comment]
This issue still exists

**Comment by LawrenceBeran** [comment]
I found I have the same problem and I had a go at fixing the issue.

The problem is that the directory parameter of the Kodi JSONRPC call for VideoLibrary.Scan expects a folder but Radarr is passing it the movie path.

I have worked out a fix for this by just trimming the filename from the movie path. This doesn't work though in the case the movie folder changes though.

I am a bit of a github noob and cannot work out how to push my branch (either that or I do not have the rights).

I added the following line:
`moviePath = new OsPath(moviePath).Directory.FullPath.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);`

Into JsonApiProvider.cs

        private void UpdateMovieLibrary(XbmcSettings settings, Movie movie)
        {
            try
            {
                var moviePath = GetMoviePath(settings, movie);

                if (moviePath != null)
                {
                    moviePath = new OsPath(moviePath).Directory.FullPath.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
                    _logger.Debug("Updating movie {0} (Path: {1}) on XBMC host: {2}", movie, moviePath, settings.Address);
                }

To get this to work for changing movie folders would be more work though, currently the event handler is passing a Movie object with the old movie path (Radarr path) rather than the updated path. The logic to work out if the path has changed is harder as the Radarr path root can be different to the Kodi path root as well as other path elements.

The code would be able to handle it better if it had both the original and updated Radarr movie path names. Or in the case of a path change just call VideoLibrary.Scan without the directory parameter.

I hope the patch I have done at least helps a bit for this excellent project.

Regards
